lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -105,4 +114,31 @@ public function getLastResponse($force = false)
 
         return ($force) ? $lastResponse : $this->lastCachedResponse;
     }
+
+    /**
+     * Format the path and add query parameters if they exist.
+     *
+     * @param string $path
+     * @param array  $options
+     * @return void
+     */
+    private function formatPath($path, array $options)
+    {
+        $this->path = $path;
+
+        if (array_key_exists('query', $options) && !empty($options['query'])) {
+            $this->path .= '?';
+
+            $i = 0;
+            foreach ($options['query'] as $key => $value) {
```

Comment:
Why do you want to encode/urlencode? I still don't get it.

How do you manage it when `$path` contains only what is given with the custom query syntax `$client->getHttpClient()->get("repos/jrean/askjong.com/commits?sha=master&path=articles/hello-world.md")`? It already contains query string parameters not encoded.